src/tests/Common/Directory.Build.targets

```diff
@@ -36,6 +37,14 @@
   <Target Name="CopyDependencyToCoreRoot"
           DependsOnTargets="ResolveAssemblyReferences;ResolveRuntimeFilesFromLocalBuild">
 
+    <!-- Publish crossgen2 on supported platforms. -->
+
+    <MSBuild Condition="'$(Crossgen2Supported)' == 'true'"
+      Targets="Restore;PublishToDisk"
+      BuildInParallel="true"
+      Properties="NativeAotSupported=$(NativeAotSupported);OutputPath=$(CORE_ROOT)\crossgen2"
+      Projects="$(InstallerProjectRoot)pkg\sfx\Microsoft.NETCore.App\Microsoft.NETCore.App.Crossgen2.sfxproj" />
```

Comment:
Let's follow the same pattern elsewhere in the repo where we use a separate invocation with `MSBuildRestoreSessionId` set to a new Guid to ensure we get the correct restore experience.